### ISSUE TITLE
Conversion script falling over SABnzbd

### ISSUE THREAD

**Issue by yekul** [mdhiggins/sickbeard_mp4_automator#1337]
Previous issue completely resolved, and for some reason SABnzbd has now begun to fall over. It just seemed to stop working, and i'm not too sure why. This was on an older build prior to new branch being added so doubt that's the issue. I updated to ensure no older problems, but nothing improved. 

This is the output i'm receiving: 
```

> 2020-10-04 15:13:12 - resources.mediaprocessor - INFO - Starting conversion.
> 2020-10-04 15:13:12 - resources.mediaprocessor - INFO - FFmpeg command:
> 2020-10-04 15:13:12 - resources.mediaprocessor - INFO - ======================
> 2020-10-04 15:13:12 - resources.mediaprocessor - INFO - C:\FFmpeg\bin\ffmpeg.exe -hwaccel dxva2 -i C:\Users\server\Downloads\complete\Brassic.S01E04.Episode.4.1080p.AMZN.WEB-DL.DD+2.0.H.264-Cinefeel\cd63ae5d98499b002713b092debe9c03.mkv.original -vcodec None -map 0:0 -field_order progressive -crf 20 -metadata:s:v BPS=6580000 -metadata:s:v BPS-eng=6580000 -metadata:s:v title=FHD -c:a:0 ac3 -map 0:1 -ac:a:0 2 -b:a:0 256k -metadata:s:a:0 BPS=256000 -metadata:s:a:0 BPS-eng=256000 -metadata:s:a:0 title=Stereo -metadata:s:a:0 language=eng -disposition:a:0 +default-dub-original-comment-lyrics-karaoke-forced-hearing_impaired-visual_impaired-captions -f matroska -threads 0 -metadata:g encoding_tool=SMA -y C:\Users\server\Downloads\complete\Brassic.S01E04.Episode.4.1080p.AMZN.WEB-DL.DD+2.0.H.264-Cinefeel\cd63ae5d98499b002713b092debe9c03.mkv
> 2020-10-04 15:13:12 - resources.mediaprocessor - INFO - ======================
> 0%
> 2020-10-04 15:13:12 - resources.mediaprocessor - ERROR - Unexpected exception during conversion.
> Traceback (most recent call last):
> File "C:\Users\server\Downloads\sickbeard_mp4_automator-master\resources\mediaprocessor.py", line 1487, in convert
> for timecode, debug in conv:
> File "C:\Users\server\Downloads\sickbeard_mp4_automator-master\converter\__init__.py", line 308, in convert
> for timecode, debug in self.ffmpeg.convert(outfile, optlist,
> File "C:\Users\server\Downloads\sickbeard_mp4_automator-master\converter\ffmpeg.py", line 688, in convert
> cmd = ' '.join(cmds)
> TypeError: sequence item 6: expected str instance, NoneType found
> 2020-10-04 15:13:12 - SABPostProcess - ERROR - Converting file failed C:\Users\server\Downloads\complete\Brassic.S01E04.Episode.4.1080p.AMZN.WEB-DL.DD+2.0.H.264-Cinefeel\cd63ae5d98499b002713b092debe9c03.mkv.
> 2020-10-04 15:13:12 - SABPostProcess - ERROR - No valid files found for conversion in download, aborting.
> 2020-10-04 15:13:12 - SABPostProcess - ERROR - Unexpected exception.
> Traceback (most recent call last):
> File "C:\Users\server\Downloads\sickbeard_mp4_automator-master\SABPostProcess.py", line 81, in <module>
> sys.exit(1)
> SystemExit: 1
```

Updated FFmpeg as well just incase. Attaching relevant section below incase there's anything odd. Only thing I changed recently was adding aac to audio codec however, and it was working fine after that briefly. 

> ffmpeg = C:\FFmpeg\bin\ffmpeg.exe
> ffprobe = C:\FFmpeg\bin\ffprobe.exe
> threads = 0
> hwaccels = dxva2, cuvid, qsv, d3d11va, vaapi
> hwaccel-decoders = h264_cuvid, mjpeg_cuvid, mpeg1_cuvid, mpeg2_cuvid, mpeg4_cuvid, vc1_cuvid, hevc_qsv, h264_qsv, hevc_vaapi, h264_vaapi
> hwdevices = vaapi:/dev/dri/renderD128
> hwaccel-output-format = vaapi:vaapi
> output-directory = 
> output-format = mkv
> output-extension = mkv
> temp-extension = 
> minimum-size = 0
> ignored-extensions = nfo, ds_store
> copy-to = 
> move-to = 
> delete-original = True
> sort-streams = True
> process-same-extensions = True
> force-convert = False
> post-process = False
> detailed-progress = True
> preopts = 
> postopts = 
> wait-post-process = False

> [Video]
codec = hevc, h265, x265
max-bitrate = 0
crf = 20
crf-profiles = 
codec-parameters = 
max-width = 0
profile = 
max-level = 0.0
pix-fmt = 
filter = 
force-filter = False
postopts = -preset,veryfast
dynamic-parameters = False
preset = 

> [Audio]
codec = ac3, aac
languages = eng
default-language = eng
first-stream-of-language = True
allow-language-relax = False
channel-bitrate = 128
max-bitrate = 0
max-channels = 0
prefer-more-channels = True
default-more-channels = True
filter = 
force-filter = False
sample-rates = 
copy-original = False
aac-adtstoasc = False
ignore-truehd = 
ignored-dispositions = 
unique-dispositions = False

> [Universal Audio]
codec = ac3
channel-bitrate = 128
first-stream-only = False
move-after = False
filter = 
force-filter = False

> [SABNZBD]
convert = True
sickbeard-category = sickbeard
sickrage-category = sickrage
couchpotato-category = couchpotato
sonarr-category = sonarr
radarr-category = radarr
bypass-category = bypass
output-directory = 
path-mapping =

**Comment by yekul** [comment]
Hrm, this is also happening for my previously working qbittorrent setup you added the path mapping options in for earlier. Which is a completely different install and script being used inside the VM. Gives the same error code. 

When trying to do a manual.py test I am getting the same errors as well.

**Comment by mdhiggins** [comment]
Set you video codec option to
`codec = h265, x265, hevc`

**Comment by mdhiggins** [comment]
Actually, that script should have accounted for that
9db826be5eac062447058c2a854e90e5f01d46ac

This update should fix your old way of doing things

This was always a weird issue since ffmpeg uses h265 as its coding naming convention but ffprobe reports hevc. I added wrapper classes that tried to make both work but the wrapper for hevc -> h265 was inheriting wrong

Now both naming conventions should be valid

**Comment by yekul** [comment]
Oh man you're totally right I kept thinking nothing changed and it just suddenly stopped working. But I believe it was because I swapped it round from 'h265, x265, hevc' to 'hevc, h265, x265' after some previous discussion. 

It's currently running the post process script right now so hopefully good to go again. Sorry for all the hassles... 

Also worth noting, any plans to push https://github.com/mdhiggins/sickbeard_mp4_automator/issues/1332 into main? My SABnzbd also only shows '===============' unless I replace the code you mentioned above. Running 3.0.2 SABnzbd. Will make myself a note for updates to reapply the fix if it will cause issues for other users. 

Closing this as it appears good to go again thanks once again...
